tests/test_xyz/test_points_poly.py

```diff
@@ -234,229 +78,172 @@ def test_points_in_polygon():
-
-    fence = pol.get_fence(
-        distance=10, nextend=1, name="SOMENAME", asnumpy=False, atleast=3
-    )
-    logger.info(fence.dataframe.to_string())
-
-    assert len(fence.dataframe) == 5
-    assert fence.dataframe.H_DELTALEN.mean() == pytest.approx(12.49, abs=0.02)
-    assert fence.dataframe.H_DELTALEN.std() <= 0.001
-
-
-def test_rename_columns():
-    """Renaming xname, yname, zname"""
-
-    pol = Polygons(POLSET2)
-    assert pol.xname == "X_UTME"
-
-    pol.xname = "NEWX"
-    assert pol.xname == "NEWX"
-
-    assert "NEWX" in pol.dataframe
-
-    pol.yname = "NEWY"
-    assert pol.yname == "NEWY"
-    assert pol.xname != "NEWY"
-
-    assert "NEWY" in pol.dataframe
-
-
-def test_empty_polygon_has_default_name():
-    pol = Polygons()
-    assert pol.dtname == "T_DELTALEN"
-
-
-def test_check_column_names():
-    pol = Polygons()
-    data = pd.DataFrame({"T_DELTALEN": [1, 2]})
-    pol.dataframe = data
-    assert pol.dtname == "T_DELTALEN"
-    assert pol.dhname is None
-
-
-def test_delete_from_empty_polygon_does_not_fail(recwarn):
-    pol = Polygons()
-    pol.delete_columns(pol.dtname)
-    assert len(recwarn) == 1
-    assert "Trying to delete" in str(recwarn.list[0].message)
+@pytest.mark.parametrize(
+    "oper, expected", [("add", 12), ("sub", 8), ("div", 5), ("mul", 20), ("set", 2)]
+)
+def test_oper_single_point_in_polygon(oper, expected):
+    pol = Polygons(SMALL_POLY_INNER)
+    poi = Points([(4.0, 4.0, 10.0)])
+    # operators work on z-values
+    poi.operation_polygons(pol, value=2, opname=oper, inside=True)
+    assert poi.dataframe[poi.zname].values[0] == expected
 
 
-@pytest.mark.parametrize("test_name", [(1), ({}), ([]), (2.3)])
 @pytest.mark.parametrize(
-    "name_attribute", [("hname"), ("dhname"), ("tname"), ("dtname"), ("pname")]
+    "oper, expected",
+    [
+        ("add", [12, 14, 12, 10]),
+        ("sub", [8, 6, 8, 10]),
+        ("div", [5, 2.5, 5, 10]),
+        ("mul", [20, 40, 20, 10]),
+        ("set", [2, 2, 2, 10]),
+    ],
 )
-def test_raise_incorrect_name_type(test_name, name_attribute):
-    pol = Polygons()
-    data = pd.DataFrame({"ANYTHING": [1, 2]})
-    pol.dataframe = data
-
-    with pytest.raises(ValueError) as execinfo:
-        setattr(pol, name_attribute, test_name)
-    assert "Wrong type of input" in str(execinfo.value)
+def test_oper_points_outside_overlapping_polygon(oper, expected):
+    pol = Polygons(SMALL_POLY_INNER + SMALL_POLY_OVERLAP_INNER)
+    # The Four points are placed: within first poly, within the overlap, within the
+    # second poly, outside both poly
+    poi = Points(
+        [
+            (3.5, 3.5, 10.0),
+            (4.5, 4.5, 10.0),
+            (5.5, 5.5, 10.0),
+            (6.5, 6.5, 10.0),
+        ]
+    )
+    # Operations are performed n times, where n reflects the number of polygons a
+    # point is within
+    poi.operation_polygons(pol, value=2, opname=oper, inside=True)
+    assert list(poi.dataframe[poi.zname].values) == expected
```

Comment:
Multiple polygons will result in operations performed multiple times for points in the intersection of polygons (if `inside=True`), or outside the union (if `inside=False`)

I would have expected that the operations are performed only once though, but this is intended behaviour?